go/check_tag.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+# Used in go CI on tagged workflows.
+# Checks that the current commit is tagged with the correct MCAP library version.
+set -eo pipefail
+
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 <path-to-mcap-binary>"
+    exit 1
+fi
+
+expected_tag="go/mcap/$($1 version --library)"
+read -ra all_tags <<< "$(git tag --points-at HEAD)"
+found="false"
+for tag in "${all_tags[@]}"; do
+    if [ "$tag" = "$expected_tag" ]; then
+        found="true"
+    fi
+done
```

Comment:
Might be slightly simpler if you used `git rev-parse go/mcap/$(...)` and compared with `git rev-parse HEAD`.